packages/node/src/edge-functions/edge-handler.mts

```diff
@@ -165,31 +184,25 @@ async function createEdgeRuntimeServer(params?: {
 
     const server = await runServer({ runtime });
 
-    const onExit = async () => {
-      // When exiting this process, wait for the Edge Runtime server to finish
-      // all its work, especially waitUntil promises before exiting this process.
-      //
-      // Here we use a short timeout (10 seconds) to let the user know that
-      // it has a long-running waitUntil promise.
-      const WAIT_UNTIL_TIMEOUT = 10 * 1000;
-      const waitUntil = server.close();
-      return new Promise<void>((resolve, reject) => {
+    // @ts-expect-error
+    runtime.context.globalThis[Symbol.for('@vercel/request-context')] = {
+      get: () => ({
+        waitUntil: params.awaiter.waitUntil.bind(params.awaiter),
+      }),
+    };
```

Comment:
what happens if we have two requests concurrently? do we need AsyncLocalStorage?

Reply:
can we use Object.defineProperty here too?